### ISSUE TITLE
Move io.psrfits tests to their own directory

### ISSUE THREAD

**Issue by mhvk** [mhvk/scintillometry#134]
While it seems logical to have all tests of pulsar tasks in `scintillometry.tasks`, the `io` are really rather stand-alone and may as well be moved (after all, `psrfits` might make sense as part of baseband!).

**Comment by luojing1211** [comment]
PSRFITSReader class is a "BaseTaskBase" type, should I change it to some other types? If we want to move the io-test to another directory, pytest recommends us to do something like:
```
setup.py
src/
   scintillometry/
        __init__.py
        dedispersion.py
        fold.py
tests/
    __init__.py
    io/
        __init__.py
        test_psrfits_io.py
    tasks/
        __init__.py
        test_tasks.py
```

**Comment by mhvk** [comment]
Don't worry about the type yet - we might well end up merging those bases between `baseband` and `scintillometry` at some point.

As for test locations, I actually like having the test directory closer to the source, ie., to follow astropy and baseband and have them in `io/psrfits/tests`.

**Comment by luojing1211** [comment]
Ok, the inlining test should work as well.

**Comment by mhvk** [comment]
fixed by #138
